packages/fether-react/src/utils/withAccount.js

```diff
@@ -5,18 +5,45 @@
 
 import React from 'react';
 import { withRouter } from 'react-router-dom';
+import { compose, mapProps } from 'recompose';
+import light from '@parity/light.js-react';
+
+import { transactionCountOf$, withoutLoading } from '@parity/light.js';
+import withAccountsInfo from '../utils/withAccountsInfo';
 
 const AccountAddressFromRouter = withRouter(props =>
   props.children(props.match.params.accountAddress)
 );
 
-// We don't want to pass the router props to the returned component
-export default Component => initialProps => {
-  return (
-    <AccountAddressFromRouter>
-      {accountAddress => (
-        <Component accountAddress={accountAddress} {...initialProps} />
-      )}
-    </AccountAddressFromRouter>
-  );
-};
+// We do not want to pass the router props nor the accountsInfo props, both
+// used internally, down to the component returned by withAccount.
+export default Component =>
+  withAccountsInfo(({ accountsInfo, ...initialProps }) => {
+    return (
+      <AccountAddressFromRouter>
+        {accountAddress => {
+          const DecoratedComponent = compose(
+            light({
```

Comment:
```js
            light({
              transactionCount: () =>
                transactionCountOf$(accountAddress).pipe(withoutLoading())
            }),
```
This makes the component remount on every new block (blank screen for a short period then TxForm appears again, reset). Not sure how to rewrite this?
If I remove those lines the TxForm fields just reset on every block (even though there is no remounting)

Reply:
I believ HOCs in the render method is an anti-pattern.

Would something like this work?
```javascript
const WithAccount = compose(
  withRouter,
  withAccountsInfo,
  light({
    transactionCount: props =>
      transactionCountOf$(props.router.accountAddress).pipe(withoutLoading())
  }),
  mapProps(...)
)(props => props.children(props.account));

export default Component => initialProps => (
  <WithAccount>
    {account => <Component {...initialProps} account={account} />}
  </WithAccount>
);
```

Reply:
it does work! much cleaner :+1: still doesn't resolve the reset issue though, I'll continue investigating

Reply:
actually my bad, it does seem to have resolved the issue!! :tada: